packages/core/src/assertEvent.ts

```diff
@@ -0,0 +1,23 @@
+import { EventObject, ExtractEvent } from './types';
+
+export function assertEvent<
+  TEventType extends TEvent['type'],
+  TEvent extends EventObject = EventObject
+>(e: TEvent, type: TEventType): asserts e is ExtractEvent<TEvent, TEventType>;
+export function assertEvent<
+  TEventType extends TEvent['type'],
+  TEvent extends EventObject = EventObject
+>(e: TEvent, type: TEventType[]): asserts e is ExtractEvent<TEvent, TEventType>;
```

Comment:
1. Would using a single overload with `e: TEvent | TEvent[]` change something here? for now I'm just curious, not requesting any changes
2. I would prefer replacing `e` with `event` - this stuff pops up in the IDE tooltips and it would be best for this to avoid abbreviations
3. I would, personally, reorder those generics. I feel like it's easier to make a mistake if we use the proposed order because `assertEvent<'FOO'>` will use the `EventObject` since partial inference is not supported by TS. I understand that the optional type param can't be used first but I think it should be OK to just drop that default here.

Reply:
> Would using a single overload with e: TEvent | TEvent[] change something here? for now I'm just curious, not requesting any changes

I thought to try this too. For some mystic eldritch reasons, it does not work with that definition.

> I would prefer replacing e with event - this stuff pops up in the IDE tooltips and it would be best for this to avoid abbreviations

Done

> I would, personally, reorder those generics. I feel like it's easier to make a mistake if we use the proposed order because assertEvent<'FOO'> will use the EventObject since partial inference is not supported by TS. I understand that the optional type param can't be used first but I think it should be OK to just drop that default here.

Done